plane/src/database/backend.rs

```diff
@@ -435,7 +424,10 @@ impl<'a> BackendDatabase<'a> {
     }
 
     pub async fn publish_metrics(&self, metrics: BackendMetricsMessage) -> sqlx::Result<()> {
-        emit_ephemeral_with_key(&self.db.pool, &metrics.backend_id.to_string(), &metrics).await
+        let mut txn = self.db.pool.begin().await?;
+        emit_ephemeral_with_key(&mut txn, &metrics.backend_id.to_string(), &metrics).await?;
+        txn.commit().await?;
+        Ok(())
```

Comment:
The `emit_*` functions now take a `PgConnection`. The easiest way to get that from a pool is to create a transaction.